flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskAsyncCallTest.java

```diff
@@ -277,8 +287,12 @@ public void invoke() throws Exception {
 				}
 			}
 
-			triggerLatch.trigger();
 			if (error != null) {
+				// exit method prematurely due to error but make sure that the tests can finish
+				triggerLatch.trigger();
```

Comment:
for all latches, it should also have:`if (!latch.isTriggered()) { latch.await() }`

Reply:
Why is that? I think at this point the latch might not get triggered at all.

Reply:
Sorry, I was just looking on the IDE and missed the lines. This line should be before every time you call `await` on the `latch`.

Reply:
I think it doesn't matter because the latch already checks for the flag.
```
	public void await() throws InterruptedException {
		synchronized (lock) {
			while (!triggered) {
				lock.wait();
			}
		}
	}
```

Reply:
yes, a latch that was already triggered will simply return immediately, no need for an additional check